runtime/smithy-test/common/src/aws/smithy/kotlin/runtime/smithy/test/HttpResponseTest.kt

```diff
@@ -88,7 +88,16 @@ fun <T> httpResponseTest(block: HttpResponseTestBuilder<T>.() -> Unit) = runSusp
             }
 
             val body: HttpBody = testBuilder.expected.body?.let {
-                ByteArrayContent(it.encodeToByteArray())
+                // emulate a real response stream which typically can only be consumed once
+                // see: https://github.com/awslabs/aws-sdk-kotlin/issues/356
+                object : HttpBody.Streaming() {
+                    private var consumed = false
+                    override fun readFrom(): SdkByteReadChannel {
+                        val content = if (consumed) ByteArray(0) else it.encodeToByteArray()
+                        consumed = true
+                        return SdkByteReadChannel(content)
+                    }
+                }
             } ?: HttpBody.Empty
```

Comment:
**Question**: While I recognize that we were missing how consume-once streams were handled before, it seems like we'll now miss how replayable streams and simple byte arrays are handled. Is there value in testing both cases?

Reply:
Response bodies are _always_ going to be a single use stream. At least starting from the engine. What happens in middleware may change this (as we do with wrapped response).